petastorm/spark/spark_dataset_converter.py

```diff
@@ -183,6 +226,38 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
         self.reader.join()
 
 
+class TorchDatasetContextManager(object):
+    """
+    A context manager that manages the creation and termination of a
+    :class:`petastorm.Reader`.
+    """
+
+    def __init__(self, data_url, batch_size, num_epochs, workers_count,
+                 cur_shard, shard_count, **petastorm_reader_kwargs):
+        """
+        :param data_url: A string specifying the data URL.
+        See `SparkDatasetConverter.make_torch_dataloader()` for the definitions
+        of the other parameters.
+        """
+        from petastorm.pytorch import DataLoader
+
+        petastorm_reader_kwargs["num_epochs"] = num_epochs
+        if workers_count is not None:
+            petastorm_reader_kwargs["workers_count"] = workers_count
+        petastorm_reader_kwargs["cur_shard"] = cur_shard
```

Comment:
need a util function to check whether this is running as an MPI environment, warn users if MPI_COMM_SIZE is not 1.